airflow/www/app.py

```diff
@@ -971,6 +971,10 @@ def action(self):
             # Flagging tasks as successful
             session = settings.Session()
             task_ids = [task_id]
+            end_date = ((dag.latest_execution_date or datetime.now())
+                        if future else execution_date)
+            start_date = dag.default_args['start_date'] if past else execution_date
```

Comment:
`dag.default_args['start_date']` may not exist
Maybe `min([task.start_date for task in dag.tasks])`

I think we should limit the number of runs though, and maybe alert if reached.